invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/bulk_actions/SearchResultsBulkActionsManager.js

```diff
@@ -0,0 +1,82 @@
+import { BulkActionsContext } from "./context";
+import React, { Component } from "react";
+import Overridable from "react-overridable";
+import _hasIn from "lodash/hasIn";
+
+class SearchResultsBulkActionsManager extends Component {
+  constructor(props) {
+    super(props);
+
+    this.selected = {};
+    this.state = { allSelected: false, selectedCount: 0 };
+  }
+
+  addToSelected = (rowId) => {
+    const { selectedCount } = this.state;
+
+    if (_hasIn(this.selected, `${rowId}`)) {
+      this.selected[rowId] = !this.selected[rowId];
+    } else {
+      this.selected[rowId] = true;
+    }
+
+    if (!this.selected[rowId]) {
+      this.setAllSelected(false);
+      this.setSelectedCount(selectedCount-1);
+    } else {
+      const updatedCount = selectedCount+1;
+      this.setSelectedCount(updatedCount);
+      if(Object.keys(this.selected).length === updatedCount){
+        this.setAllSelected(true);
+      }
+    }
+  };
+
+  setSelectedCount = (count) =>{
+    this.setState({selectedCount: count})
+  }
+
+  setAllSelected = (val, global = false) => {
```

Comment:
I don't understand what is happening here, and `global` is scary :)
Can you try to improve naming to make it more readable?